spec/std/raise_spec.cr

```diff
@@ -0,0 +1,25 @@
+require "spec"
+
+struct CallStack # allow clone and equality
+  def_clone
+  def_equals @callstack
+end
+
+describe "raise" do
+  it "should throw exception with existing callstack if already set" do
+    ex = Exception.new "with callstack"
+    ex.callstack = CallStack.new
+    callstack_to_match = ex.callstack.clone
+    new_ex = expect_raises Exception, "with callstack" do
+      raise ex
+    end
+    new_ex.callstack.should eq(callstack_to_match)
+  end
+
+  it "should throw exception with new callstack if not already set" do
```

Comment:
Maybe `it sets exception's callstack` and it should come before the spec that says `it doesn't overwrite the callback`.